FindKqueue.cmake

```diff
@@ -12,8 +12,7 @@ if (NOT HAVE_KQUEUE)
         find_path(LIBKQUEUE_ROOT_DIR NAMES "include/sys/event.h")
 
         # Prefer linking statically but look for a shared library version too.
-        find_library(LIBKQUEUE_LIBRARIES NAMES libkqueue.a libkqueue.so
-                     HINTS ${LIBKQUEUE_ROOT_DIR}/lib)
+        find_library(LIBKQUEUE_LIBRARIES NAMES kqueue HINTS ${LIBKQUEUE_ROOT_DIR}/lib)
```

Comment:
I've tested a bit locally on Ubuntu 24.04 with libkqueue-dev installed system-wide.

With the following patch, I can do `-D LIBKQUEUE_ROOT_DIR:string=/usr/include` to build against that.
```
diff --git a/FindKqueue.cmake b/FindKqueue.cmake
index 01e5a7d..7e53683 100644
--- a/FindKqueue.cmake
+++ b/FindKqueue.cmake
@@ -14,13 +14,16 @@ if (NOT HAVE_KQUEUE)
         # Prefer linking statically but look for a shared library version too.
         find_library(LIBKQUEUE_LIBRARIES NAMES kqueue HINTS ${LIBKQUEUE_ROOT_DIR}/lib)
 
-        find_path(LIBKQUEUE_INCLUDE_DIRS NAMES "sys/event.h"
-                  HINTS ${LIBKQUEUE_ROOT_DIR}/include/kqueue)
+        find_path(LIBKQUEUE_INCLUDE_DIRS NAMES "kqueue/sys/event.h"
+                  HINTS ${LIBKQUEUE_ROOT_DIR}/include)
 
         include(FindPackageHandleStandardArgs)
         find_package_handle_standard_args(LIBKQUEUE DEFAULT_MSG LIBKQUEUE_LIBRARIES
                                           LIBKQUEUE_INCLUDE_DIRS)
 
+        # Append /kqueue so that including <sys/event.h> directly works.
+        set(LIBKQUEUE_INCLUDE_DIRS "${LIBKQUEUE_INCLUDE_DIRS}/kqueue")
+
         mark_as_advanced(LIBKQUEUE_ROOT_DIR LIBKQUEUE_LIBRARIES LIBKQUEUE_INCLUDE_DIRS)
```

It does use `libkqueue.so`, however and the Zeek executable is dynamically linked against the system version (which would totally be fine IMO).
```
$ ldd $(which zeek) | grep kq
        libkqueue.so.0 => /lib/x86_64-linux-gnu/libkqueue.so.0 (0x00007fa4bfbeb000)

```
However, some btests fail and I see we have a private patch for libkqueue, so this seems all moodt Approving again :-)

Reply:
> -D LIBKQUEUE_ROOT_DIR:string=/usr/include

Shouldn't that be `/usr`, since the library is in `/usr/lib` and the headers are in `/usr/include`? I just installed the same package on my ubuntu 22.04 machine here and that's how it's laid out. With `/usr` it doesn't require any changes to the include dir variables, but it does still link against the shared library. Maybe that's fine?

The reason it opts for the static library in the bundled build is that we explicitly tell it to: https://github.com/zeek/cmake/blob/master/FindKqueue.cmake#L122-L124. It's the same case for c-ares: https://github.com/zeek/cmake/blob/master/FindCAres.cmake#L41

I'm inclined to merge this as-is since it fixes a bug, but open a ticket to talk about whether we should opt for static vs shared on the external builds.